src/pages/Transactions/transactionUtils.tsx

```diff
@@ -31,70 +29,92 @@ const isUserRejectionError = (error: any) => {
     withdrawal_credentials,
-    // eslint-disable-next-line camelcase
+    signature,
+    // eslint-disable-next-line
     deposit_data_root,
-  } = depositFile;
-
-  try {
-    updateTransactionStatus(pubkey, TransactionStatus.PENDING);
-    const walletProvider: any = await (connector as AbstractConnector).getProvider();
-    const web3: Eth = new Web3(walletProvider).eth;
-    const contract = new web3.Contract(contractAbi, CONTRACT_ADDRESS);
-    const transactionParameters: SendOptions = {
-      gasPrice: '0x0055e72a000', // TODO estimate gas value
-      from: account as string,
-      value: TX_VALUE,
-    };
+  } = depositFiles[recursionIdx];
 
-    contract.methods
-      .deposit(
-        prefix0X(pubkey),
-        prefix0X(withdrawal_credentials),
-        prefix0X(signature),
-        prefix0X(deposit_data_root)
-      )
-      .send(transactionParameters)
-      .on('transactionHash', (txHash: string): void => {
-        updateTransactionStatus(pubkey, TransactionStatus.STARTED, txHash);
-      })
-      .on('receipt', () => {
-        // do something?
-      })
-      .on(
-        'confirmation',
-        (confirmation: number, receipt: { status: {} }): any => {
-          if (confirmation === 0) {
-            if (receipt.status) {
-              updateTransactionStatus(pubkey, TransactionStatus.SUCCEEDED);
-            } else {
-              updateTransactionStatus(pubkey, TransactionStatus.FAILED);
-            }
+  contract.methods
+    .deposit(
+      prefix0X(pubkey),
+      prefix0X(withdrawal_credentials),
+      prefix0X(signature),
+      prefix0X(deposit_data_root)
+    )
+    .send(transactionParameters)
+    .on('transactionHash', (txHash: string): void => {
+      updateTransactionStatus(pubkey, TransactionStatus.STARTED, txHash);
+      handleMultipleTransactions(
+        depositFiles,
+        connector,
+        account,
+        updateTransactionStatus,
+        recursionIdx + 1
+      );
+    })
+    .on('receipt', () => {
+      // do something?
+    })
+    .on(
+      'confirmation',
+      (confirmation: number, receipt: { status: {} }): any => {
+        if (confirmation === 0) {
+          if (receipt.status) {
+            updateTransactionStatus(pubkey, TransactionStatus.SUCCEEDED);
+          } else {
+            updateTransactionStatus(pubkey, TransactionStatus.FAILED);
           }
         }
-      )
-      .on('error', (error: any) => {
-        if (isUserRejectionError(error)) {
-          updateTransactionStatus(pubkey, TransactionStatus.REJECTED);
-        } else {
-          updateTransactionStatus(pubkey, TransactionStatus.FAILED);
-        }
-      });
-  } catch (rejected) {
-    updateTransactionStatus(pubkey, TransactionStatus.FAILED);
-  }
+      }
+    )
+    .on('error', (error: any) => {
+      handleMultipleTransactions(
+        depositFiles,
+        connector,
+        account,
+        updateTransactionStatus,
+        recursionIdx + 1
+      );
```

Comment:
```suggestion
```

If the user rejects a TX or something goes wrong, we should not continue presenting more TXs. Instead the user can fix the problem (or not) and continue by pressing the "Make N Deposit transactions" at the top of the page